### PR TITLE
add pprof

### PR DESCRIPTION
**Pull request by Houshoupei84**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -108,6 +111,19 @@ func setupConfig(c *cli.Context) {
...
+			listenAddr := net.JoinHostPort("", cfg.Profile)
+			log.Infof("Profile server listening on %s", listenAddr)
+			profileRedirect := http.RedirectHandler("/debug/pprof",
+				http.StatusSeeOther)
+			http.Handle("/", profileRedirect)
+			ret := http.ListenAndServe(listenAddr, nil)
+			log.Errorf("Profile server ListenAndServe return %v", ret)
+		}()
+	}
+
```
Comment (yz89) [inline main.go]: Please put the implementation in the util package and use the modularity functions.

File: utils/common.go [hunk 2]
```diff
@@ -2,9 +2,12 @@ package utils
 
 import (
 	"fmt"
-	"os"
-
 	"github.com/howeyc/gopass"
+	"net"
```
Comment (alexpan0610) [inline utils/common.go]: Format imports.

File: utils/common.go [hunk 3]
```diff
@@ -42,3 +45,17 @@ func FileExisted(filename string) bool {
...
+		listenAddr := net.JoinHostPort("", port)
+		fmt.Printf("Profile server listening on %s\n", listenAddr)
+		profileRedirect := http.RedirectHandler("/debug/pprof",
+			http.StatusSeeOther)
+		http.Handle("/", profileRedirect)
+		ret := http.ListenAndServe(listenAddr, nil)
+		fmt.Printf("Profile server ListenAndServe return %v", ret)
+	}()
+
+}
```
Comment (alexpan0610) [inline utils/common.go]: Add blank line in the end of file.

File: utils/common.go [hunk 4]
```diff
@@ -42,3 +45,17 @@ func FileExisted(filename string) bool {
 	_, err := os.Stat(filename)
 	return err == nil || os.IsExist(err)
 }
+
+func StartPProf(Profile uint32) {
```
Comment (alexpan0610) [inline utils/common.go]: Variable `Profile` should be `port`.

File: common/config/config.go [hunk 5]
```diff
@@ -58,6 +58,7 @@ type Configuration struct {
 	VoteStartHeight    uint32            `json:"VoteStartHeight"`
 	CRCOnlyDPOSHeight  uint32            `json:"CRCOnlyDPOSHeight"`
 	PublicDPOSHeight   uint32            `json:"PublicDPOSHeight"`
+	Profile            uint32            `json:"Profile"`
```
Comment (alexpan0610) [inline common/config/config.go]: Change `Profile` to `ProfilePort`.

**Pull request merged**